vscode-dotnet-runtime-library/src/EventStream/TelemetryObserver.ts

```diff
@@ -15,30 +21,105 @@ export interface ITelemetryReporter {
-        } else if (event.isError) {
-            this.telemetryReporter.sendTelemetryErrorEvent(event.eventName, properties);
-        } else {
-            this.telemetryReporter.sendTelemetryEvent(event.eventName, properties);
+    /**
+     *
+     * @param event The event posted to the event stream that we subscribed to.
+     * @remarks The TelemetryReporter from the VSCode library contains all of the logic when we try to send an event for us.
+     * It will handle whether we should send the telemetry or not. However, it is not aware of our custom telemetry logic
+     * So we might as well check both to be safe.
+     *
+     * However, the .NET SDK itself has its own telemetry and we must keep track of VS Code's telemetry settings for that.
+     */
+    public post(event: IEvent): void
+    {
+        if(TelemetryObserver.isTelemetryEnabled(this.isExtensionTelemetryEnabled))
+        {
+            const properties = event.getSanitizedProperties(); // Get properties that don't contain personally identifiable data
+            if (!properties) {
+                this.telemetryReporter.sendTelemetryEvent(event.eventName);
+            } else if (event.isError) {
+                this.telemetryReporter.sendTelemetryErrorEvent(event.eventName, properties);
+            } else {
+                this.telemetryReporter.sendTelemetryEvent(event.eventName, properties);
+            }
         }
     }
 
-    public dispose(): void {
+    public dispose(): void
+    {
         this.telemetryReporter.dispose();
     }
+
+    static setDotnetSDKTelemetryToMatch(isExtensionTelemetryEnabled : boolean, extensionContext : vscode.ExtensionContext, eventStream : IEventStream)
+    {
+        if(!TelemetryObserver.isTelemetryEnabled(isExtensionTelemetryEnabled))
+        {
+            TelemetryObserver.logTelemetryChange(`Before disabling .NET SDK telemetry:`, isExtensionTelemetryEnabled, eventStream);
+
+            new CommandExecutor(eventStream).setEnvironmentVariable(
+                'DOTNET_CLI_TELEMETRY_OPTOUT',
+                'true',
+                extensionContext.environmentVariableCollection,
+
+`Telemetry is disabled for the .NET Install Tool, but we were unable to turn off the .NET SDK telemetry.
+Please verify that .NET SDK telemetry is disabled as well by setting the environment variable DOTNET_CLI_TELEMETRY_OPTOUT to true.`,
+
+`The .NET Install Tool will not collect telemetry. However, the .NET SDK does collect telemetry.
+To disable .NET SDK telemetry, please set the environment variable DOTNET_CLI_TELEMETRY_OPTOUT to true.`);
+
+            TelemetryObserver.logTelemetryChange(`After disabling .NET SDK telemetry:`, isExtensionTelemetryEnabled, eventStream);
+        }
+        else
+        {
+            TelemetryObserver.logTelemetryChange(`Unchanged Telemetry Settings.`, isExtensionTelemetryEnabled, eventStream);
+        }
+    }
+
+    static isDotnetSDKTelemetryDisabled()

```

Comment:
Since there are 3/4 separate environments we need to edit, it may be best to put in a doc string that this is only used for logging the current process environment setting.